bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/TACmiBinding.java

```diff
@@ -205,8 +208,12 @@ protected void execute() {
                 }
             }
 
+        } catch (SocketTimeoutException e) {
+            logger.info("Receive Timeout on CoE socket");
         } catch (Exception e) {
-            logger.warn("Error in execute: {}, Message: {}", e.getClass().getName(), e.getMessage());
+            StringWriter w = new StringWriter();
+            e.printStackTrace(new PrintWriter(w));
```

Comment:
Sorry, I don't understand the problem here. Do you in general want no stack traces in the log?

The reason why I put it there is that I got several log entries that said:

2016-02-07 14:36:57.348 [WARN ] [.b.tacmi.internal.TACmiBinding] - Error in execute: java.lang.NullPointerException, Message: null

At that point, it is not possible to find out what went wrong inside the try-block without having a stacktrace (at least for me). Thats why I want one here. Unfortunately the NPE did not happen again since then, so I was not able to dig deeper into the problem.

So if there is a guideline not to put stack traces to the log, thats fine for me. But where should they go then?

In general I'm not a friend of silently eating up exceptions, because it makes it difficult to correct the problems that they would show.